build_runner/lib/src/daemon/daemon_builder.dart

```diff
@@ -204,17 +206,16 @@ class BuildRunnerDaemonBuilder implements DaemonBuilder {
               // Assume we will create an outputDir.
               true,
               expectedDeletes,
-            ));
-
-    var changes = graphEvents
+            ))
         .map((data) => WatchEvent(data.type, '${data.id}'))
         .map((change) => [change]);
 
     var changeProvider = daemonOptions.buildMode == BuildMode.Auto
```

Comment:
Should we pull out all this into a `_createChangeProvider` function? Or maybe move it into the constructor for `AutoChangeProvider`?

Reply:
I don't want to move it all into the constructor I don't think (it is a good thing imo that it just accepts a stream of changes right now). I think it is probably fine as is - I could move the inline function to an actual instance method but I don't know that there is really any value there (and I would have more boilerplate for passing in params etc even though its only used locally in this function).

Reply:
Could also consider skipping the local function and moving the whole chunk of code inline where we call the constructor - I'm not sure how ugly that would make things with a large chunk like that in a ternary conditional...

Reply:
Ya that gets pretty ugly :/

I also did try extracting it to a separate method but that ends up needing 6 parameters so its pretty boilerplaty compared to what I have now.